modules/core/src/main/java/org/apache/ignite/internal/causality/OutdatedTokenException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.causality;
+
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+import org.apache.ignite.lang.IgniteStringFormatter;
+
+/**
+ * This exception is thrown when getting a value from {@link VersionedValue#get(long)} with a token outnound of stored history
+ * {@link VersionedValue#historySize}.
+ * <p>
+ * {@link VersionedValue} stores a value per the causality token.
+ * See {@link VersionedValue#get(long)}.
+ */
+public class OutdatedTokenException extends IgniteInternalCheckedException {
+
+    /**
+     * Constructor.
+     *
+     * @param token The tokent which is expired.
+     */
+    public OutdatedTokenException(long token) {
```

Comment:
Perhaps, it makes sense to add an additional constructor(s) that will accept the following parameters:
 - Optional string message
 - Value of the actual token
 - Size of history

Something like as follows:
```
public OutdatedTokenException(long outdatedToken, long actualToken, int historySize);
public OutdatedTokenException(long outdatedToken, long actualToken, int historySize, String msg);
```

Reply:
I do not know how worth the message parameter in constructor required:
- Optional string message
because this exception can be thrown in only one case (the logic is encapsulated into VersionedValue).
Moreover, the exception message contains anchors and cannot be an arbitrary string.